### PR TITLE
Tools: add generator for MP parameter xml

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

**Comment by khancyr** [comment]
@Kiwa21 enjoy !

File: Tools/autotest/param_metadata/xmlemit_mp.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from emit import Emit
+from param import known_param_fields, known_units
+
+
+# Emit APM documentation in an machine readable XML format for Mission Planner
+class XmlEmitMP(Emit):
+    def __init__(self):
+        Emit.__init__(self)
+        wiki_fname = 'parametermetadata.xml'
+        self.f = open(wiki_fname, mode='w')
```
Comment (amilcarlucas) [inline Tools/autotest/param_metadata/xmlemit_mp.py]: Shouldn't this be mp_fname ?
```suggestion
        mp_fname = 'parametermetadata.xml'
        self.f = open(mp_fname, mode='w')
```

File: Tools/autotest/param_metadata/xmlemit_mp.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+from xml.sax.saxutils import escape, quoteattr
+
+from emit import Emit
+from param import known_param_fields, known_units
+
+
+# Emit APM documentation in an machine readable XML format for Mission Planner
```
Comment (amilcarlucas) [inline Tools/autotest/param_metadata/xmlemit_mp.py]:
```suggestion
# Emit Ardupilot documentation in an machine readable XML format for Mission Planner
```

File: Tools/autotest/param_metadata/xmlemit_mp.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/param_metadata/xmlemit_mp.py]: done

**Comment by khancyr** [comment]
address comments and fix issue

**Comment by amilcarlucas** [comment]
Does it work on both python 2 and 3 ?

**Comment by khancyr** [comment]
yep

**Comment by tridge** [comment]
@khancyr has @meee1 seen this? I'd like to know it will be used

**Comment by khancyr** [comment]
To use it, just rename it to ParameterMetaDataBackup.xml and put it into MissionPlanner directory.
It will be load on start !

**Comment by khancyr** [comment]
@meee1 could you have a look at this ! I have made the full generator and tested after sorted the current file from MP and it is conform !

**Comment by khancyr** [comment]
To use this with Mission Planner,
use ./Tools/scripts/generate_mp_paramfile.sh to generate the ParameterMetaData.xml

Copy the file in the Mission Planner directory : in /home/$USER/Mission Planner on linux, on MyDocs\Mission Planner on windows. The file will be load at start up !

**Comment by meee1** [comment]
looks ok to me, and is in the same format as what MP creates

**Comment by tridge** [comment]
please test with both py2 and py3

**Comment by khancyr** [comment]
tested and working with both Python3 and Python2 !

**Pull request merged**